test/masq.test.js

```diff
@@ -1,36 +1,85 @@
+const MasqCore = require('./MasqCore')
+// const rai = require('random-access-idb')
+// const hyperdb = require('hyperdb')
 const signalserver = require('signalhubws/server')
 const signalhub = require('signalhubws')
 const swarm = require('webrtc-swarm')
 const wrtc = require('wrtc')
 
-const Masq = require('../src/index')
+const Masq = require('../src')
 
 // user an in memory random-access-storage instead
 jest.mock('random-access-idb', () =>
   () => require('random-access-memory'))
 
+const wait = () => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      resolve()
+    }, 800)
+  })
+}
+
 let server = null
 let masq = null
-
+let masqCore = null
+let profile1 = {
+  username: 'bob',
+  image: 'image1'
+}
 beforeAll(() => {
   server = signalserver()
   server.listen(8080, () => console.log('server running'))
 })
 
-afterAll(() => {
+afterAll(async () => {
+  await wait()
   server.close()
 })
 
 test('initialize', async () => {
   masq = new Masq()
   await masq.init()
+  masqCore = new MasqCore()
+  await masqCore.init()
+  await masqCore.initProfiles()
+  await masqCore.setProfile('bob', profile1)
+  const profile = await masqCore.getProfile('bob')
+  expect(profile).toEqual(profile1)
```

Comment:
You are testing masqCore, which is a mock, this is a non-sense, be sure to focus only on the masq-lib, which is the lib the developers will integrate in their apps

Reply:
If you need to setup the Masq mock with some data, do it in a `beforeAll/beforeEach`